packages/api/utils/requestValidation.ts

```diff
@@ -29,3 +32,69 @@ export function shapeValidationForPostRequest<T>(
 
   return new SetupSuccess<T>({ portfolioDraftId }, bodyResult as unknown as T);
 }
+
+interface EnvironmentWithNoAdminProps {
+  appIndex: number;
+  envIndex: number;
+}
+interface AdministratorsFound {
+  hasPortfolioAdminRole: boolean;
+  hasAdminForEachApplication: boolean;
+  hasAdminForEachEnvironment: boolean;
+  applicationsWithNoAdmin: number[];
+  environmentsWithNoAdmin: EnvironmentWithNoAdminProps[];
+  acceptableAdministratorRoles: boolean;
+}
+
+export function findAdministrators(applicationStep: ApplicationStep): AdministratorsFound {
+  const { operators, applications } = applicationStep;
+  const hasPortfolioAdminRole = adminRoleFound(operators);
+
+  // find apps and envs without admin roles
+  const applicationsWithNoAdmin: number[] = [];
+  const environmentsWithNoAdmin: EnvironmentWithNoAdminProps[] = [];
+  applications.forEach((app, appIndex) => {
+    // apps without admin roles
+    !adminRoleFound(app.operators) && applicationsWithNoAdmin.push(appIndex);
+
+    // env without admin roles
+    app.environments.forEach((env, envIndex) => {
+      !adminRoleFound(env.operators) && environmentsWithNoAdmin.push({ appIndex, envIndex });
+    });
+  });
+
+  // if no missing admin roles in app or env, all app or env admins are considered present
+  const hasAdminForEachApplication = applicationsWithNoAdmin.length === 0;
+  const hasAdminForEachEnvironment = environmentsWithNoAdmin.length === 0;
+
+  let acceptableAdministratorRoles = hasPortfolioAdminRole || hasAdminForEachApplication || hasAdminForEachEnvironment;
+
+  if (!acceptableAdministratorRoles) {
+    let appsWithMissingAppAndEnvAdminRole = 0;
+    applicationsWithNoAdmin.forEach((appNoAdmin) => {
+      const missingEnvAdmins = environmentsWithNoAdmin.filter((obj) => obj.appIndex === appNoAdmin);
+      // if no missing env admin from apps with missing app admin, all env admins considered present
+      missingEnvAdmins.length !== 0 && appsWithMissingAppAndEnvAdminRole++;
+    });
+
+    acceptableAdministratorRoles = appsWithMissingAppAndEnvAdminRole === 0;
+  }
+
+  return {
+    hasPortfolioAdminRole,
+    hasAdminForEachApplication,
+    hasAdminForEachEnvironment,
+    applicationsWithNoAdmin,
+    environmentsWithNoAdmin,
+    acceptableAdministratorRoles,
+  };
+}
+
+export function adminRoleFound(operators: Operator[]): boolean {
```

Comment:
Can we add documentation here? I am confused about why we return `true` if _any_ admin operator is found

Reply:
Added something like the below. @kylelaker-ccpo does that help clear up the confusion? I also added documentation for the `findAdministrators` to help state what the business rules are for admin roles. 

```typescript
/**
 * Checks if at least one operator admin is present in a group of operators
 * according to the business rules.
 *
 * Only 'portfolio_administrator' operators are found at the top level, and
 * 'administrator' operators are found at the app and env levels.
 *
 * @param operators - operators at one of the three application step levels
 * @returns - true if one operator admin found, and false otherwise
 */
```

Reply:
Looks good! I think we're missing an explanation that this is because an environment/portfolio _must_ contain an administrator (which is the context I lacked)

Reply:
Understood. Thanks for asking so I could add some context to make things clearer.